sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientLiveTests.cs

```diff
@@ -102,7 +102,7 @@ await using (var producer = new EventHubProducerClient(connectionString))
 
                     // Guard against duplicates; Event Hubs has an at-least-once guarantee.
 
-                    if ((TryAdd(processedEvents, eventId, args.Data)) && (Interlocked.Increment(ref processCount) >= sentCount))
+                    if ((processedEvents.TryAdd(eventId, args.Data)) && (Interlocked.Increment(ref processCount) >= sentCount))
```

Comment:
Non blocking - I believe the interlocked logic doesn't need to be used here.  I think that We could just look at the `Count` property of `processedEvents`, since it wouldn't be increased in the case where TryAdd does not add a duplicate to the dictionary?

Reply:
Yeah... that would make sense, wouldn't it.   Now that we've moved to `ConcurrentDictionary` we definitely don't need it.    Probably should have revisited that after we went from "I just need a count" to "I should track each event."    😄